request/pool.go

```diff
@@ -88,20 +90,31 @@ func (rp *Pool) start() {
 	rp.pending.Start()
 }
 
+func (rp *Pool) addRandomnessToFirstStrike() time.Duration {
```

Comment:
Lets make sure that in the worse case the first strike is still > BatchTimeout. IE compare the BatchTimeout with the FirstStrikeThreshold. It seems to me the FirstStrikeThreshold should be at least 5*BatchTimeout or so...

Even without randomization, if FirstStrikeThreshold < BatchTimeout + network-delay we'll get constant forwarding. So some sanity checks are required.

Reply:
Will add checks 
Currently the batch timeout is 500ms and we are running tests with first strike of 2s